module/modmanager/manager.go

```diff
@@ -67,15 +70,15 @@ type module struct {
 	// pendingRemoval allows delaying module close until after resources within it are closed
 	pendingRemoval bool
 
-	// inRecovery stores whether or not an OnUnexpectedExit function is trying
-	// to recover a crash of this module; inRecoveryLock guards the execution of
+	// inStartup stores whether or not an OnUnexpectedExit function is trying
+	// to recover a crash of this module; inStartupLock guards the execution of
 	// an OnUnexpectedExit function for this module.
 	//
 	// NOTE(benjirewis): Using just an atomic boolean is not sufficient, as OUE
 	// functions for the same module cannot overlap and should not continue after
 	// another OUE has finished.
-	inRecovery     atomic.Bool
-	inRecoveryLock sync.Mutex
+	inStartup     atomic.Bool
+	inStartupLock sync.Mutex
```

Comment:
I think this should still be called `inRecoveryLock`, it's only relevant for OUEs and stops them from overlapping with each other. Isn't used in `add`/during startup.

Reply:
ahh good call! got too greedy with `s/inRecovery/inStartup` :)